src/main/java/org/apache/sling/rewriter/impl/ProcessorManagerImpl.java

```diff
@@ -327,61 +326,14 @@ private synchronized void updateProcessor(final String path) {
                 }
                 if ( index != -1 ) {
                     // we are already in the array
-                    if ( index == 0 ) {
-                        // we are the first, so remove the old, and add the new
                         this.orderedProcessors.remove(configs[index].config);
                         configs[index] = new ConfigEntry(path, config);
                         if ( config.isActive() ) {
                             this.orderedProcessors.add(config);
                             Collections.sort(this.orderedProcessors, new ProcessorConfiguratorComparator());
-                        }
-                    } else {
```

Comment:
Removed this clause we can't just replace the item, we have also update this.orderedProcessors...